File: src/main/java/me/ag2s/epublib/domain/Author.java
package me.ag2s.epublib.domain;

import java.io.Serializable;
import me.ag2s.epublib.util.StringUtil;

public class Author implements Serializable {
   private static final long serialVersionUID = 6663408501416574200L;
   private String firstname;
   private String lastname;
   private Relator relator;

   public Author(String singleName) {
      this("", singleName);
   }

   public Author(String firstname, String lastname) {
      this.relator = Relator.AUTHOR;
      this.firstname = firstname;
      this.lastname = lastname;
   }

   public String getFirstname() {
      return this.firstname;
   }

   public void setFirstname(String firstname) {
      this.firstname = firstname;
   }

   public String getLastname() {
      return this.lastname;
   }

   public void setLastname(String lastname) {
      this.lastname = lastname;
   }

   public String toString() {
      return this.lastname + ", " + this.firstname;
   }

   public int hashCode() {
      return StringUtil.hashCode(this.firstname, this.lastname);
   }

   public boolean equals(Object authorObject) {
      if (!(authorObject instanceof Author)) {
         return false;
      } else {
         Author other = (Author)authorObject;
         return StringUtil.equals(this.firstname, other.firstname) && StringUtil.equals(this.lastname, other.lastname);
      }
   }

   public void setRole(String code) {
      Relator result = Relator.byCode(code);
      if (result == null) {
         result = Relator.AUTHOR;
      }

      this.relator = result;
   }

   public Relator getRelator() {
      return this.relator;
   }

   public void setRelator(Relator relator) {
      this.relator = relator;
   }
}

File: src/main/kotlin/me/ag2s/epublib/domain/Author.kt
package me.ag2s.epublib.domain

import java.io.Serializable
import me.ag2s.epublib.util.StringUtil

class Author(firstname: String?, lastname: String?) : Serializable {
    private var firstname = firstname
    private var lastname = lastname
    private var relator = Relator.AUTHOR
    constructor(singleName: String?) : this("", singleName)
    fun getFirstname() = firstname; fun setFirstname(value: String?) { firstname = value }
    fun getLastname() = lastname; fun setLastname(value: String?) { lastname = value }
    override fun toString(): String = "$lastname, $firstname"
    override fun hashCode(): Int = StringUtil.hashCode(firstname, lastname)
    override fun equals(other: Any?): Boolean = other is Author && StringUtil.equals(firstname, other.firstname) && StringUtil.equals(lastname, other.lastname)
    fun setRole(code: String?) { relator = Relator.byCode(code) ?: Relator.AUTHOR }
    fun getRelator() = relator; fun setRelator(value: Relator) { relator = value }
    companion object { private const val serialVersionUID = 6663408501416574200L }
}
